netease/content/contents/code/netease.js

```diff
@@ -121,7 +123,10 @@ var NeteaseResolver = Tomahawk.extend( api_to_extend, {
         return this._apiCall('search/get', {type:1, s:query, limit:100}).then(function(results){
             if(!results.result)
                 results = JSON.parse(results);
-            return results.result.songs.map(that._convertTrack, that);
+            if(results.result.songCount > 0)
+                return results.result.songs.map(that._convertTrack, that);
+            else
+                return [];
```

Comment:
I prefer to have { } all the time.. avoids nasty bugs and makes sprintf debugging easier.:p  (I know I merged the rest ... so you dont need to change but I'd appreciate it)